src/lib/journey/callbacks/redirect/redirect.svelte

```diff
@@ -0,0 +1,34 @@
+<script lang="ts">
+  import { FRAuth, FRStep, type RedirectCallback } from '@forgerock/javascript-sdk';
+
+  import type {
+    CallbackMetadata,
+    SelfSubmitFunction,
+    StepMetadata,
+  } from '$journey/journey.interfaces';
+  import Spinner from '$components/primitives/spinner/spinner.svelte';
+  import Text from '$components/primitives/text/text.svelte';
+  import type { Style } from '$lib/style.store';
+  import type { Maybe } from '$lib/interfaces';
+
+  export let callback: RedirectCallback;
+  export let callbackMetadata: CallbackMetadata;
+  export let selfSubmitFunction: Maybe<SelfSubmitFunction> = null;
+  export let stepMetadata: StepMetadata;
+  export let style: Style = {};
+
+  export let step: FRStep;
```

Comment:
Injecting the step into the component felt slightly wrong since we don't do this for any other callbacks, however I needed access to it for the SDK call to work `FRAuth.redirect() `expects the entire step rather than just the callback. If there's a nicer way to achieve this I'm open to ideas.

Reply:
Okay, this is a good callout. Let's definitely avoid this as it breaks the Rule of Least Privilege. Passing a callback component the whole step is giving it too much rope. Let's move the actual redirection into the stage component as it manages the data at the step level. At the end of the day, the rendering of the UI and the redirecting of the browser are separate things, and don't need to be coupled.

Reply:
This makes more sense. I've moved the redirect into the stage and a utility function. Let me know what you think.

Reply:
Nice, I think this works.